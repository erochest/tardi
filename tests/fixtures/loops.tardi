use: std/vectors

// initialize
{ }
0

[
  // skip 2
  dup 2 == [
    1 +
    continue
  ] when

  // stop 5
  dup 5 == [
    // clean up
    drop
    break
  ] when

  // push and increment
  2dup swap push!
  1 +
] loop

// leave something so we know it's executing things after the loop
"still here"

0
[ dup 5 < ] [
    1 +
] while
