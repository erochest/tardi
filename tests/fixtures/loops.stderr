{ 0 1 3 4 }
"still here"
5
